File: MedicalSampleDistribution/Frontend/src/types/distribution.ts
export interface Distribution {
  id: number;
  representativeCode: number;
  representativeName: string;
  supervisorCode: number;
  supervisorName: string;
  materialId: string;
  materialDescription: string;
  importId: number;
  importDate: string | null;
  importState: string | null;
  legajoSap: string | null;
  cant: number;
}

export interface DistributionListResponse {
  distributions: Distribution[];
  totalItems: number;
  totalPages: number;
  currentPage: number;
}

export interface DistributionSummary {
  representativeCode: number;
  representativeName: string;
  totalQuantity: number;
  materialsCount: number;
}

export interface RepresentativeDistribution {
  representativeCode: number;
  representativeName: string;
  materials: MaterialDistribution[];
  totalQuantity: number;
}

export interface MaterialDistribution {
  materialId: string;
  materialDescription: string;
  quantity: number;
}

File: MedicalSampleDistribution/Backend/MMDistributionSystem.API/Distribution.cs
using System;
using System.Collections.Generic;

namespace MMDistributionSystem.API;

public partial class Distribution
{
    public int Id { get; set; }

    public int RepresentativeCode { get; set; }

    public int SupervisorCode { get; set; }

    public string MaterialId { get; set; } = null!;

    public int ImportId { get; set; }

    public string? LegajoSap { get; set; }

    public int Cant { get; set; }

    public virtual Import Import { get; set; } = null!;
}
